pkg/k8sutil/create.go

```diff
@@ -157,3 +172,59 @@ func parseJSONManifest(data []byte) ([]manifest, error) {
 	}
 	return manifests, nil
 }
+
+// CreateOrUpdateNamespace creates the release namespace or updates the namespace
+// if it already exists.
+func CreateOrUpdateNamespace(ns Namespace, nsclient corev1typed.NamespaceInterface) error {
+	if ns.Name == "" {
+		return fmt.Errorf("namespace name can't be empty")
+	}
+
+	namespace, err := nsclient.Get(context.TODO(), ns.Name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return createNamespace(ns, nsclient)
+		}
+
+		return fmt.Errorf("getting namespace %q: %w", ns.Name, err)
+	}
+	// Namespace exists, hence updating the namespace.
+	return updateNamespace(namespace, ns, nsclient)
+}
+
+// updateNamespace updates an existing namespace.
+func updateNamespace(namespace *v1.Namespace, ns Namespace, nsclient corev1typed.NamespaceInterface) error {
+	// Merge new labels and annotations with existing ones.
+	updatedLabels := internal.MergeMaps(ns.Labels, namespace.ObjectMeta.Labels)
+	updatedAnnotations := internal.MergeMaps(ns.Annotations, namespace.ObjectMeta.Annotations)
+
+	namespace.ObjectMeta.Labels = updatedLabels
```

Comment:
Small suggestion not asking for change and it's totally up to you, as `updatedLabels` and `updatedAnnotations` are not used anywhere, you could directly assign it.

Reply:
it makes the lines very long with direct assignment.

Reply:
Cool